### ISSUE TITLE
Add cases modal on Scenario page saves duplicates

### ISSUE THREAD

**Issue by macfarlandian** [Recidiviz/covid19-dashboard#594]
**Describe the bug**
When adding historical data to a facility from the Scenario page (facilities list), saving cases creates two versions of that set of inputs in the facility's `modelVersions` collection

**To Reproduce**

1. Go to multifacility page for any scenario
2. Click on a case count to open the modal
3. enter some data and save it
4. See error: two identical copies (minus a tiny difference in updatedAt) of that day's data appear in Firestore

**Expected behavior**
Only one copy should be saved

**Screenshots**
Screenshots revealing sensitive information _must not_ be shared. If a screenshot is helpful in diagnosing and resolving the bug, please send this directly to a contact on the Recidiviz team.

**Desktop (please complete the following information):**

- OS [e.g. Windows 10]: macOS Catalina
- Browser [e.g. chrome, edge]: Firefox
- Version [e.g. 22]: latest

**Smartphone (please complete the following information):**
n/a

**Additional context**
The cause appears to be [the save handler](https://github.com/Recidiviz/covid19-dashboard/blob/master/src/page-multi-facility/MultiFacilityImpactDashboard.tsx#L89) that is passed into the add cases modal on that page; it executes the `FacilitiesContext`'s `createOrUpdateFacility` action, which is already performed by the modal component. There isn't any meaningful callback we need to execute here, so this function should become an optional prop on the modal or be removed entirely.

**Comment by jessex** [comment]
@macfarlandian if I understand correctly, this has no actual impact on user-facing functionality, as far as we know, because the app layer will pick the more "recent" of the two duplicate observations when passing observation series to the R(t) function or when preparing state for the Add Cases modal, right?

If so, we should still fix it for the sake of good hygiene but it's probably not ultra high priority.

CC @sychang

**Comment by sychang** [comment]
This one is timely because it may become obsolete after an upcoming change to make r(t) take a 7-day rolling average that will let us get rid of the duplicate hack.

cc @justkunz

**Comment by macfarlandian** [comment]
@jessex yes that's right, it's creating and storing useless data but not affecting anything (these duplicates are always filtered out before they get to the Rt function anyway ... to @sychang 's point they are same-day duplicates, not duplicate case counts across days, so it won't be affected by that change one way or another)
